### PR TITLE
*: make binlog and clustered index exclusive

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Problem Summary:

Since the binlog does not support clustered index tables, we should make them exclusive.

### What is changed and how it works?

What's Changed:
- When the binlog is enabled, users cannot build clustered index table even if the "CLUSTERED" keyword is specified.
- For the existing clustered index tables, the mutations of the tables will not write to the binlog.
- The global variable `@@tidb_enable_clustered_index` is hidden from `show variables`.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test
- Integration test

Side effects

NA

### Release note 

- The changes to clustered index tables will not be written to the binlog. When the binlog is enabled, all the newly created tables are non-clustered.

### CODE REVIEW

**Comment by ngaut** [comment]
Looks like we should update release notes.

**Comment by zz-jason** [comment]
+1, It's a big compatibility change, please write the behavior in the release note section.

File: table/tables/tables.go [hunk 1]
```diff
@@ -999,7 +999,7 @@ func (t *TableCommon) RemoveRecord(ctx sessionctx.Context, h kv.Handle, r []type
 		return err
 	}
 
-	if shouldWriteBinlog(ctx) {
+	if shouldWriteBinlog(ctx) && !t.meta.IsCommonHandle {
```
Comment (coocood) [inline table/tables/tables.go:1002]: How about we change `shouldWriteBinlog` to add the table meta as an argument?
Then we will be always consistent.

**Pull request merged**